### PR TITLE
Create Components-of-an-api.md

### PR DESCRIPTION
**Pull request by johnwinifred**
Content for API course

### CODE REVIEW

File: content/docs/API docs course/Components-of-an-api.md [hunk 1]
```diff
@@ -0,0 +1,658 @@
...
+        },
+        "Content-Type": {
+          "description": "The MIME type of the body of the request",
+          "schema": {
+            "type": "string"
+          },
+          "example": "application/json"
+        }
+      }
+    }
```
Comment (wise4rmgod) [inline content/docs/API docs course/Components-of-an-api.md:209]: Please use a markdown code syntax here.
Example
```
Code here
```

File: content/docs/API docs course/Components-of-an-api.md [hunk 2]
```diff
@@ -0,0 +1,658 @@
...
+3. **API key:** An API key is a unique identifier issued by the API providers themselves. When making a request to a server that uses an API key, you have to include the API key as a parameter in the request body. 
+
+
+4. **Basic Oauth:** This is a simple authentication method that allows you to send HTTP requests by providing a username and password in plain text. 
+****
+
+
+**API Authentication in our** **Sample Petstore API**
+
+
```
Comment (wise4rmgod) [inline content/docs/API docs course/Components-of-an-api.md:298]: Add Markdown code syntax here too

File: content/docs/API docs course/Components-of-an-api.md [hunk 3]
```diff
@@ -0,0 +1,658 @@
...
+An API endpoint is the location where the request by the API client is sent to retrieve the resource from the server. An application can have multiple endpoints. Each of these endpoints retrieves a specific resource. Since API endpoints lead to the specific location of the resource, we use them to test the API response. 
+
+
+**API Endpoints in our** **Sample Petstore API**
+
+
+    /pets
+    
+    /pets/{petId}
```
Comment (wise4rmgod) [inline content/docs/API docs course/Components-of-an-api.md:349]: Add Markdown code syntax here too

File: content/docs/API docs course/Components-of-an-api.md [hunk 4]
```diff
@@ -0,0 +1,658 @@
...
+
+The **GET** method is used to request data from a specified resource. From our Sample Petstore API, you can use the `GET` method to retrieve a list of pets or get details about a specific pet.
+
+**Endpoint:** `/pets`
+**Description:** Fetches a list of all pets in the store. You can optionally limit the number of pets returned using a query parameter.
+
+**Example Request:**
+
+
+    GET https://api.petstore.example.com/v1/pets?limit=10
```
Comment (wise4rmgod) [inline content/docs/API docs course/Components-of-an-api.md:379]: Add Markdown code syntax here too

File: content/docs/API docs course/Components-of-an-api.md [hunk 5]
```diff
@@ -0,0 +1,658 @@
...
+        "id": 1,
+        "name": "Doggo",
+        "tag": "dog"
+      },
+      {
+        "id": 2,
+        "name": "Kitty",
+        "tag": "cat"
+      }
+    ]
```
Comment (wise4rmgod) [inline content/docs/API docs course/Components-of-an-api.md:397]: Add Markdown code syntax here too

File: content/docs/API docs course/Components-of-an-api.md [hunk 6]
```diff
@@ -0,0 +1,658 @@
...
+      "id": 3,
+      "name": "Bunny",
+      "tag": "rabbit"
+    }
+
+
+E**xample Response:**
+
+
+    {
```
Comment (wise4rmgod) [inline content/docs/API docs course/Components-of-an-api.md:426]: Add Markdown code syntax here too

File: content/docs/API docs course/Components-of-an-api.md [hunk 7]
```diff
@@ -0,0 +1,658 @@
...
+
+The `PUT` method is used to update an existing resource or create a new resource if it does not already exist.
+
+**Endpoint:** `/pets/{petId}`
+**Description:** Updates the details of an existing pet by its ID.
+
+**Example Request:**
+
+
+    PUT https://api.petstore.example.com/v1/pets/1
```
Comment (wise4rmgod) [inline content/docs/API docs course/Components-of-an-api.md:444]: Add Markdown code syntax here too

File: content/docs/API docs course/Components-of-an-api.md (hunk 1 above)
Comment (johnwinifred) [reply content/docs/API docs course/Components-of-an-api.md:209]: Done

File: content/docs/API docs course/Components-of-an-api.md (hunk 3 above)
Comment (johnwinifred) [reply content/docs/API docs course/Components-of-an-api.md:349]: Done

File: content/docs/API docs course/Components-of-an-api.md (hunk 4 above)
Comment (johnwinifred) [reply content/docs/API docs course/Components-of-an-api.md:379]: Done

File: content/docs/API docs course/Components-of-an-api.md [hunk 8]
```diff
@@ -0,0 +1,658 @@
...
+The **GET** method is used to request data from a specified resource. From our Sample Petstore API, you can use the `GET` method to retrieve a list of pets or get details about a specific pet.
+
+**Endpoint:** `/pets`
+**Description:** Fetches a list of all pets in the store. You can optionally limit the number of pets returned using a query parameter.
+
+**Example Request:**
+
+
+    GET https://api.petstore.example.com/v1/pets?limit=10
+    X-API-Key: your-api-key-here
```
[the remark this answers is not in the document]
Comment (johnwinifred) [reply content/docs/API docs course/Components-of-an-api.md:380]: Done

File: content/docs/API docs course/Components-of-an-api.md (hunk 5 above)
Comment (johnwinifred) [reply content/docs/API docs course/Components-of-an-api.md:398]: Done

File: content/docs/API docs course/Components-of-an-api.md [hunk 9]
```diff
@@ -0,0 +1,658 @@
...
+
+    POST https://api.petstore.example.com/v1/pets
+    X-API-Key: your-api-key-here
+    Content-Type: application/json
+    
+    {
+      "id": 3,
+      "name": "Bunny",
+      "tag": "rabbit"
+    }
```
[the remark this answers is not in the document]
Comment (johnwinifred) [reply content/docs/API docs course/Components-of-an-api.md:421]: Done

File: content/docs/API docs course/Components-of-an-api.md (hunk 6 above)
Comment (johnwinifred) [reply content/docs/API docs course/Components-of-an-api.md:427]: Done

File: content/docs/API docs course/Components-of-an-api.md (hunk 7 above)
Comment (johnwinifred) [reply content/docs/API docs course/Components-of-an-api.md:445]: Done

**Pull request merged**